### PR TITLE
refactor: Use ranges

### PR DESCRIPTION
**Pull request by slavek-kucera**

### CODE REVIEW

File: parser_library/test/lexing/logical_line_test.cpp [hunk 1]
```diff
@@ -267,9 +267,12 @@ TEST(logical_line, iterator_difference)
     const auto e_1 = std::prev(e);
 
     EXPECT_EQ(std::distance(b, e), e - b);
+    EXPECT_EQ(std::ranges::distance(b, e), e - b);
     EXPECT_EQ(-(b - e), e - b);
     EXPECT_EQ(std::distance(b, l1e), l1e - b);
+    EXPECT_EQ(std::ranges::distance(b, l1e), l1e - b);
     EXPECT_EQ(-(b - l1e), l1e - b);
     EXPECT_EQ(std::distance(b, e_1), e_1 - b);
+    EXPECT_EQ(std::ranges::distance(b, e_1), e_1 - b);
```
Comment (jirimosinger) [inline parser_library/test/lexing/logical_line_test.cpp:276]: Is it intentional to have both `std::distance` and `std::ranges::distance` tests? Is it to verify that it works correctly for both algorithms?

File: parser_library/test/lsp/lsp_context_instr_test.cpp [hunk 2]
```diff
@@ -63,13 +63,14 @@ struct lsp_context_instr : public ::testing::Test
 };
 
 namespace {
+constexpr std::string_view ADDFRR = "ADDFRR";
 auto label_addfrr_compare = [](const auto& item) { return item.label == "ADDFRR"; };
```
Comment (jirimosinger) [inline parser_library/test/lsp/lsp_context_instr_test.cpp:67]: This is probably not used anymore

```suggestion
```

File: parser_library/test/lsp/lsp_context_instr_test.cpp [hunk 3]
```diff
@@ -87,9 +88,9 @@ TEST_F(lsp_context_instr, ADDFRR_loaded_changed_instr_set)
     auto comp_list_z15 = get_completion_list(instruction_set_version::Z15);
     auto comp_list_xa = get_completion_list(instruction_set_version::XA);
 
-    auto result_z15 = std::none_of(comp_list_z15.begin(), comp_list_z15.end(), label_addfrr_compare);
+    auto result_z15 = std::ranges::find(comp_list_z15, ADDFRR, &completion_item::label) == comp_list_z15.end();
 
-    auto result_xa = std::any_of(comp_list_xa.begin(), comp_list_xa.end(), label_addfrr_compare);
+    auto result_xa = std::ranges::find(comp_list_xa, "ADDFRR", &completion_item::label) != comp_list_xa.end();
```
Comment (jirimosinger) [inline parser_library/test/lsp/lsp_context_instr_test.cpp:93]:
```suggestion
    auto result_xa = std::ranges::find(comp_list_xa, ADDFRR, &completion_item::label) != comp_list_xa.end();
```

File: parser_library/test/lexing/logical_line_test.cpp (hunk 1 above)
Comment (slavek-kucera) [reply parser_library/test/lexing/logical_line_test.cpp:276]: Yes, it is intentional, to test both algs.

**Review by jirimosinger — APPROVED** [review]

**Pull request merged**